code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -527,9 +529,50 @@ var/global/list/damage_icon_parts = list()
 		face_standing += rgb(,,,120)
 
 	overlays_standing[HAIR_LAYER]	= image(face_standing)
-
 	if(update_icons)   update_icons_layers()
 
+/mob/living/carbon/human/update_eyes(var/update_icons=1)
+	if(QDESTROYING(src))
+		return
+
+	//Reset our eyes
+	overlays_standing[EYES_LAYER]	= null
+
+	//This is ONLY for glowing eyes for now. Boring flat eyes are done by the head's own proc.
+	if(!species.has_glowing_eyes)
+		if(update_icons) update_icons_layers()
+		return
+
+	//Our glowy eyes should be hidden if some equipment hides them.
+	if(!should_have_organ(O_EYES) || (head && (head.flags_inv & BLOCKHAIR)) || (wear_mask && (wear_mask.flags_inv & BLOCKHAIR)))
+		if(update_icons) update_icons_layers()
+		return
+
+	//Get the head, we'll need it later.
+	var/obj/item/organ/external/head/head_organ = get_organ(BP_HEAD)
+	if(!head_organ || head_organ.is_stump() )
+		if(update_icons)   update_icons_layers()
+		return
+
+	//The eyes store the color themselves, funny enough.
+	var/obj/item/organ/internal/eyes/eyes = internal_organs_by_name[O_EYES]
+	if(!head_organ.eye_icon)

```

Comment:
Looks like this if statement can be eliminated by adding `|| !head_organ.eye_icon` to line 553.

Reply:
I'll change both of...